### PR TITLE
Enabling XML documentation for the Rules project.

### PR DESCRIPTION
**Pull request by RobGallo**
#### Describe the change

This will provide intellisense information for users of our NuGet package.

- Updated missing class and method documentation.
- Fixed invalid formatting in XML
- Removed a resource file which was not used.

#### PR checklist

- [ ] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/master/docs/Scenarios.md) completed?
- [ ] Does this address an existing issue? If yes, Issue# - 
- [ ] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [ ] Attach any screenshots / GIF's that are applicable.

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

File: src/AccessibilityInsights.Rules/EvaluationCode.cs [hunk 1]
```diff
@@ -8,12 +8,42 @@ namespace Axe.Windows.Rules
...
     public enum EvaluationCode
     {
-        NotApplicable, // given element did not meet the conditions for the rule
+        /// <summary>
+        /// The given element did not meet the conditions for the rule and was not evaluated.
+        /// </summary>
+        NotApplicable,
+
+        /// <summary>
+        /// The given element met the success criteria  of the rule evaluation.
```
Comment (karanbirsingh) [inline src/AccessibilityInsights.Rules/EvaluationCode.cs]: There is an extra whitespace character between 'criteria' and 'of the rule evaluation'

File: src/AccessibilityInsights.Rules/EvaluationCode.cs [hunk 2]
```diff
@@ -8,12 +8,42 @@ namespace Axe.Windows.Rules
...
+        /// </summary>
         Pass,
-        Error, // fail
-        Open, // not enough information to make a determination
-        Note, // informational
+
+        /// <summary>
+        /// The given element did not meet the success criteria  of the rule evaluation.
```
Comment (karanbirsingh) [inline src/AccessibilityInsights.Rules/EvaluationCode.cs]: There is an extra whitespace character between 'criteria' and 'of the rule evaluation'

File: src/AccessibilityInsights.Rules/EvaluationCode.cs [hunk 3]
```diff
@@ -8,12 +8,42 @@ namespace Axe.Windows.Rules
...
+        /// </summary>
+        Open,
+
+        /// <summary>
+        /// The rule's description is informational and does not indicate success or failure.
+        /// </summary>
+        Note,
+
+        /// <summary>
+        /// The given element did not meet the success criteria  of the rule evaluation,
```
Comment (karanbirsingh) [inline src/AccessibilityInsights.Rules/EvaluationCode.cs]: There is an extra whitespace character between 'criteria' and 'of the rule evaluation'

File: src/AccessibilityInsights.Rules/RunResult.cs [hunk 4]
```diff
@@ -4,11 +4,30 @@
...
 {
+    /// <summary>
+    /// Contains the result of the evaluation by a rule of a specific element.
+    /// </summary>
     public class RunResult
     {
+        /// <summary>
+        /// Metadata about the rule that was run.
+        /// </summary>
         public RuleInfo RuleInfo{ get; set; }
```
Comment (karanbirsingh) [inline src/AccessibilityInsights.Rules/RunResult.cs]: While you are here, do you mind adding a whitespace character between the RuleInfo property name and its opening bracket?

File: src/AccessibilityInsights.Rules/Rules.csproj [hunk 5]
```diff
@@ -319,11 +316,6 @@
       <Generator>ResXFileCodeGenerator</Generator>
       <LastGenOutput>LocalizedControlTypeNames.Designer.cs</LastGenOutput>
     </EmbeddedResource>
-    <EmbeddedResource Include="Resources\Messages.resx">
```
Comment (karanbirsingh) [inline src/AccessibilityInsights.Rules/Rules.csproj]: I would like to better understand the purpose of the removed Messages.resx file. Do the new documentation files make the old Messages.resx file obsolete?

File: src/AccessibilityInsights.Rules/CCAControlTypesFilter.cs [hunk 6]
```diff
@@ -6,6 +6,9 @@
 
 namespace Axe.Windows.Rules
 {
+    /// <summary>
+    /// Remove this class from Rules assembly
```
Comment (karanbirsingh) [inline src/AccessibilityInsights.Rules/CCAControlTypesFilter.cs]: Want to double-check the added comment here about removing the class from the Rules assembly. Would it be better to omit this and track the follow-up in a different way?

**Comment by DaveTryon** [comment]
> }

Line 52 should be indented one level

---
Refers to: src/AccessibilityInsights.Rules/CCAControlTypesFilter.cs:52 in 3cbffff. [](commit_id = 3cbffff0522ab342d7ba1e4473ade6a06e582469, deletion_comment = False)

File: src/AccessibilityInsights.Rules/RuleInfo.cs [hunk 7]
```diff
@@ -7,15 +7,50 @@
...
+        /// A description of the conditions under which a rule will be evaluated.
+        /// This information is generated programatically when a rule inherits the Rule base class
+        /// and is not meant to be changed.
+        /// </summary>
         public string Condition { get; set;  }
 
+        /// <summary>
+        /// Provides a string summary of the information contained in the RuleInfo object.
+        /// </summary>
+        /// <returns></returns>
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Rules/RuleInfo.cs]: >returns [](start = 13, length = 7)

Can the empty returns block be removed?

File: src/AccessibilityInsights.Rules/Rules.cs [hunk 8]
```diff
@@ -27,6 +33,12 @@ public static class Rules
             return dictionary;
         }
 
+        /// <summary>
+        /// Run a single rule based on the given RuleID.
+        /// </summary>
+        /// <param name="id"></param>
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Rules/Rules.cs]: >param [](start = 13, length = 5)

Should we have meaningful comments for id, element, and returns?

File: src/AccessibilityInsights.Rules/Rules.cs [hunk 9]
```diff
@@ -44,6 +56,11 @@ public static RunResult RunRuleByID(RuleId id, IA11yElement element)
             return retVal;
         }
 
+        /// <summary>
+        /// Run all the rules in the Rules assembly.
+        /// </summary>
+        /// <param name="element"></param>
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Rules/Rules.cs]: >param [](start = 13, length = 5)

Should we have meaningful comments for element and returns?

File: src/AccessibilityInsights.Rules/RunResult.cs [hunk 10]
```diff
@@ -4,11 +4,30 @@
...
         public IA11yElement element { get; set; }
+
+        /// <summary>
+        /// The result of the evaluation by the rule.
+        /// </summary>
         public EvaluationCode EvaluationCode { get; set; }
+
+        /// <summary>
+        /// If <see cref="EvaluationCode"/> contains the value <see cref="EvaluationCode.RuleExecutionError"/>,
+        /// this property may contain a description of the error.
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Rules/RunResult.cs]: >may [](start = 26, length = 3)

If no description is contained, is it null or string.Empty?

File: src/AccessibilityInsights.Rules/Conditions/ValueCondition.cs [hunk 11]
```diff
@@ -8,7 +8,7 @@
 namespace Axe.Windows.Rules
 {
     /// <summary>
-    /// This class enables comparisons such as ==, !=, >, and < in conditions.
+    /// This class enables comparisons such as ==, !=, &gt;, and &lt; in conditions.
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Rules/Conditions/ValueCondition.cs]: >&gt; [](start = 55, length = 4)

&gt; and &lt; will look right in intellisense but look odd here. Is there a way to make it work in both cases?

File: src/AccessibilityInsights.Rules/PropertyConditions/ElementGroups.cs [hunk 12]
```diff
@@ -78,8 +78,8 @@ private static Condition CreateUnfocusableControlsBasedOnOfficeCondition()
 
         private static Condition CreateExpectedNotToBeFocusableCondition()
         {
-            /// originally, menu was the part of this list. 
-            /// however based on feedback from Guru(Trust tester) and MSDN, menu is now counted as keyboard focusable element. 
+            // originally, menu was the part of this list. 
+            // however based on feedback from Guru(Trust tester) and MSDN, menu is now counted as keyboard focusable element.
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Rules/PropertyConditions/ElementGroups.cs]: >Guru [](start = 46, length = 4)

Should we have names in the code?

**Comment by RobGallo** [comment]
When I looked in the file, there were no messages. In addition, the files removal did not cause a compilation error, meaning it was not used anywhere. So I'm guessing it should have been removed before and just wasn't.

File: src/AccessibilityInsights.Rules/RunResult.cs (hunk 4 above)
Comment (RobGallo) [reply src/AccessibilityInsights.Rules/RunResult.cs]: Soitently

File: src/AccessibilityInsights.Rules/Rules.csproj (hunk 5 above)
Comment (RobGallo) [reply src/AccessibilityInsights.Rules/Rules.csproj]: h

> > }
> 
> Line 52 should be indented one level
> 
> Refers to: src/AccessibilityInsights.Rules/CCAControlTypesFilter.cs:52 in [3cbffff](https://github.com/Microsoft/accessibility-insights-windows/commit/3cbffff0522ab342d7ba1e4473ade6a06e582469). [](commit_id = [3cbffff](https://github.com/Microsoft/accessibility-insights-windows/commit/3cbffff0522ab342d7ba1e4473ade6a06e582469), deletion_comment = False)

Somehow I can't see which file you're refering to.

File: src/AccessibilityInsights.Rules/Rules.cs (hunk 8 above)
Comment (RobGallo) [reply src/AccessibilityInsights.Rules/Rules.cs]: I think additional comments would be both redundant and superfluous. But perhaps I'm not imagining how these parameters could be misinterpretted. Suggestions?

File: src/AccessibilityInsights.Rules/Rules.cs (hunk 9 above)
Comment (RobGallo) [reply src/AccessibilityInsights.Rules/Rules.cs]: Same response as above question.

File: src/AccessibilityInsights.Rules/Conditions/ValueCondition.cs (hunk 11 above)
Comment (RobGallo) [reply src/AccessibilityInsights.Rules/Conditions/ValueCondition.cs]: I think we've finally found a knit too small to pick. :)

File: src/AccessibilityInsights.Rules/Rules.cs (hunk 8 above)
Comment (DaveTryon) [reply src/AccessibilityInsights.Rules/Rules.cs]: id: The Id of the rule to run
element: The element to analyze
return: The output of the analysis

(As an aside, I'd probably name id "ruleId", but that's not a big deal)

File: src/AccessibilityInsights.Rules/Conditions/ValueCondition.cs (hunk 11 above)
Comment (DaveTryon) [reply src/AccessibilityInsights.Rules/Conditions/ValueCondition.cs]: "nit", not "knit"  :)

---
In reply to: [276857109](https://github.com/Microsoft/accessibility-insights-windows/pull/320#discussion_r276857109) [](ancestors = 276857109)

**Comment by DaveTryon** [comment]
> }

The next to the last brace in CCAControlTypesFilter.cs is flush left. It should be indented 4 spaces

---
In reply to: [484700784](https://github.com/Microsoft/accessibility-insights-windows/pull/320#issuecomment-484700784) [](ancestors = 484700784)

---
Refers to: src/AccessibilityInsights.Rules/CCAControlTypesFilter.cs:52 in 3cbffff. [](commit_id = 3cbffff0522ab342d7ba1e4473ade6a06e582469, deletion_comment = False)

File: src/AccessibilityInsights.Rules/Rules.csproj (hunk 5 above)
Comment (DaveTryon) [reply src/AccessibilityInsights.Rules/Rules.csproj]: Pretty sure that Rob's response was meant for my comment in CCAControlTypesFilter.cs. Line 52 of that file is flush left and should be indented 4 spaces

**Pull request merged**